frontend/e2e/helpers.cafe.ts

```diff
@@ -1,275 +1,375 @@
+}
+
+export const createTrait = async (
+  index: number,
+  id: string,
+  value: string | boolean | number,
+) => {
+  await click('#add-trait')
+  await waitForElementVisible('#create-trait-modal')
+  await setText('[name="traitID"]', id)
+  await setText('[name="traitValue"]', `${value}`)
+  await click('#create-trait-btn')
+  await t.wait(2000)
+  await t.eval(() => location.reload())
+  await waitForElementVisible(byId(`user-trait-value-${index}`))
+  const expectedValue = typeof value === 'string' ? `"${value}"` : `${value}`
+  await assertTextContent(byId(`user-trait-value-${index}`), expectedValue)
+}
+
+export const deleteTrait = async (index: number) => {
+  await click(byId(`delete-user-trait-${index}`))
+  await click('#confirm-btn-yes')
+  await waitForElementNotExist(byId(`user-trait-${index}`))
+}
+
+const lastTestGroup = {}
+let lastTestName = undefined
+
+export const logUsingLastGroup = (message?: string) => {
+  log(undefined, message)
+}
 
 // eslint-disable-next-line no-console
-export const log = (message:string, group) => console.log('\n', '\x1b[32m', `${group ? `[${group}] ` : ''}${message}`, '\x1b[0m', '\n');
-
-export const viewFeature = async (index:number) => {
-    await click(byId(`feature-item-${index}`));
-    await waitForElementVisible('#create-feature-modal');
-};
-
-export const addSegmentOverrideConfig = async (index:number, value:string|boolean|number, selectionIndex = 0) => {
-    await click(byId('segment_overrides'));
-    await click(byId(`select-segment-option-${selectionIndex}`));
-
-    await waitForElementVisible(byId(`segment-override-value-${index}`));
-    await setText(byId(`segment-override-value-${0}`), `${value}`);
-    await click(byId('segment-override-toggle-0'));
-};
-
-export const addSegmentOverride = async (index:number, value:string|boolean|number, selectionIndex = 0, mvs:MultiVariate[] = []) => {
-    await click(byId('segment_overrides'));
-    await click(byId(`select-segment-option-${selectionIndex}`));
-    await waitForElementVisible(byId(`segment-override-value-${index}`));
-    if (value) {
-        await click(`${byId(`segment-override-${0}`)} [role="switch"]`);
-    }
-    if (mvs) {
-        await Promise.all(mvs.map(async (v, i) => {
-            await setText(`.segment-overrides ${byId(`featureVariationWeight${v.value}`)}`, `${v.weight}`);
-        }));
-    }
-};
+export const log = (group: string | undefined, message?: string) => {
+  const testName = t.test.name
+  const groupName = group ?? lastTestGroup[testName]
+
+  if (lastTestName !== testName || lastTestGroup[testName] !== groupName) {
+    const ellipsis = group === groupName ? '' : '...'
+    console.log(
+      '\n',
+      '\x1b[32m',
+      `${testName ? `${ellipsis}[${testName} tests] ` : ''}${groupName}`,
+      '\x1b[0m',
+      '\n',
+    )
+    lastTestGroup[testName] = groupName
+    lastTestName = testName
+  }
+  if (message) {
+    console.log(message)
+  }
+}
```

Comment:
Sorry, I just don't understand why this has become so much more complicated?

Reply:
It is actually a simple way to manage the logs in a concurrent environment. Tests running at the same time (in parallel) can call this function at any time, so I added a couple of things to make the logs more readable (like 3 dots ... when we display parts of a group that have already started before) and simplify the management.